web_ui/src/webworkers/watershed.worker.ts

```diff
@@ -1,186 +1,37 @@
 // Copyright (C) 2022-2025 Intel Corporation
 // LIMITED EDGE SOFTWARE DISTRIBUTION LICENSE
 
-// Dependencies get bundled into the worker
-
-import { OpenCVLoader } from '@geti/smart-tools';
-import { expose } from 'comlink';
+import { OpenCVLoader, Watershed } from '@geti/smart-tools';
+import { WatershedMethods } from '@geti/smart-tools/src/watershed/interfaces';
+import { expose, proxy } from 'comlink';
 import type OpenCVTypes from 'OpenCVTypes';
 
-import { Point } from '../core/annotations/shapes.interface';
-import { Marker } from '../pages/annotator/tools/marker-tool/marker-tool.interface';
-import { WatershedPolygon } from '../pages/annotator/tools/watershed-tool/watershed-tool.interface';
-import { approximateShape, formatContourToPoints } from './utils';
-
 declare const self: DedicatedWorkerGlobalScope;
 
-let CV: OpenCVTypes.cv | null = null;
+let opencv: OpenCVTypes;
 
-const terminate = (): void => {
-    self.close();
-};
+const waitForOpenCV = async () => {
+    if (opencv) return opencv;
 
-const waitForOpenCV = async (): Promise<boolean> => {
-    if (CV) {
-        return true;
-    } else {
-        return OpenCVLoader(self).then((cvInstance: OpenCVTypes.cv) => {
-            CV = cvInstance;
+    opencv = await OpenCVLoader();
 
-            return true;
-        });
+    if ('ready' in opencv) {
```

Comment:
i didnt know but there's a `ready` method exposed by opencv which help us in knowing if the module was fully loaded.